docs/core/MODULE.md

```diff
@@ -21,3 +21,118 @@ To help developers to create their own Otter compatible module, the Otter framew
 ## List of Otter official modules
 
 The full list of official modules can be found in the [Documentation Home page](../README.md#available-packages-and-tools).
+
+## How to create its own Otter module
+
+### Module development
+
+As indicated in the [definition session](#definition), the basic needs of an Otter module is to expose an `ng-add` schematics to be able to add the module via the command `ng add <package-name>`.
+> **Note**: Information relative to the `ng-add` schematics is available [here](https://angular.io/cli/add).
+
+As any Otter module is following the same requirement, the new module can request the installation of any other by using providing helpers that will trigger the `ng-add` schematics of the requested module.
+
+Basic example of dependency module installation:
+
+```typescript
+import { chain, noop, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { NgAddSchematicsSchema } from './schema';
+
+/**
+ * Add Otter MyModule to an Otter Project
+ *
+ * @param options
+ */
+export function ngAdd(options: NgAddSchematicsSchema): Rule {
+  return async (_tree: Tree, context: SchematicContext) => {
+    try {
+      // use dynamic import to properly raise an exception if it is not an Otter project.
+      const { applyEsLintFix, install, ngAddPackages, getO3rPeerDeps } = await import('@o3r/schematics');
+
+      // retrieve dependencies following the /^@o3r\/.*/ pattern within the peerDependencies of the current module
+      const depsInfo = getO3rPeerDeps(path.resolve(__dirname, 'path', 'to', 'package.json'));
+      return chain([
+        // optional custom action dedicated to this module
+        doCustomAction(),
+        options.skipLinter ? noop() : applyEsLintFix(),
+        // install ngx-translate and message format dependencies
+        options.skipInstall ? noop : install,
+        // add the missing Otter modules in the current project
+        ngAddPackages(depsInfo.o3rPeerDeps, { skipConfirmation: true, version: depsInfo.packageVersion, parentPackageInfo: `${depsInfo.packageName!} - setup` })
+      ]);
+    } catch (e) {
+      // If the installation is initialized in a non-Otter application, mandatory packages will be missing. We need to notify the user
+      context.logger.error(`[ERROR]: Adding MyModule has failed.
+      If the error is related to missing @o3r dependencies you need to install '@o3r/core' to be able to use the MyModule package. Please run 'ng add @o3r/core' .
+      Otherwise, use the error message as guidance.`);
+      throw (e);
+    }
+  };
+}
+```
+
+### Register your module as official module
+
+It could make sense that the developed module would be part of the modules proposed in the beginning of the Otter application.
```

Comment:
```suggestion
It could make sense that the developed module would be part of the modules proposed at the beginning of the Otter application.
```
(not sure about this one)

Reply:
Not sure neither but surely less than you so I took it :D